### PR TITLE
Add limitations

### PR DESCRIPTION
**Pull request by ryanmitchell**
@sampoyigi Can you check this over, it should be good to go now.

### CODE REVIEW

File: database/migrations/2020_10_15_000300_create_cart_restriction.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateCartRestriction extends Migration
+{
+    public function up()
+    {
+        Schema::table('igniter_coupons', function (Blueprint $table) {
+            $table->boolean('is_limited')->default(false);
```
Comment (sampoyigi) [inline database/migrations/2020_10_15_000300_create_cart_restriction.php]: From the database schema, it's hard to tell what value this column holds. Maybe `is_limited_to_cart_item` or `applies_to_cart_item`? Sorry about the back and forth ;)

**Review by sampoyigi — CHANGES_REQUESTED** [review]

**Comment by ryanmitchell** [comment]
What about something like 

has_limited_scope

?

**Comment by sampoyigi** [comment]
Still not descriptive enough.

**Comment by ryanmitchell** [comment]
Trying to avoid including item in the string as I’m conscious it may not always be limiting items - might be quantities or something else. It’s a general ‘is this scope of this coupon limited in any way’

**Comment by sampoyigi** [comment]
Yea but we should be keeping things extremely simple and Coupons are already limited by different fields in the db, adding another one without being specific creates confusion. We can always add a new field if needed.

**Comment by ryanmitchell** [comment]
Np. I’ve updated it now to is_limited_to_cart_item

File: cartconditions/Coupon.php [hunk 2]
```diff
@@ -34,51 +38,78 @@ public function getValue()
...
         catch (Exception $ex) {
             flash()->alert($ex->getMessage())->now();
             $this->removeMetaData('code');
+        }
+    }
 
+    public function beforeApply()
+    {
+        $couponModel = $this->getModel();
+        if (!$couponModel OR !$couponModel->is_limited_to_cart_item)
```
Comment (sampoyigi) [inline cartconditions/Coupon.php]: This should probably change to `if (!$couponModel OR $couponModel->is_limited_to_cart_item)`

**Review by sampoyigi — CHANGES_REQUESTED** [review]

**Pull request merged**